Don't copy items the user is not allowed to view. [2.13]

mauritsvanrees opened the pull request:
From Products.PloneHotfix20161129.

tseaver commented on src/OFS/CopySupport.py:
```diff
@@ -587,7 +587,35 @@ def _getCopy(self, container):
+                    # We do not use cp._delObject, because this would fire
+                    # events that are needless for objects that are not even in
+                    # an Acquisition chain yet.
+                    cp._delOb(k)
```
ISTM we should be logging something about the skipped / deleted objects, so that the user has at least some chance to figure out what happened to their not-quite-a-copy.

mauritsvanrees replied:
Adding a `logging.warn` line should be easy enough. I can do that.
Or are you thinking about showing something to the end user?

tseaver replied:
`logging.warn` was what I was thinking.

mauritsvanrees commented:
I have added a logger.warn line. Sample output:

"WARNING OFS While copying /Plone/public-folder, removed /Plone/public-folder/private-page from copy because user is not allowed to view the original."

Wait, we can do one better by passing the container along. Done. Sample output:

"WARNING OFS While copying /Plone/public-folder to /Plone/maurits-folder, removed /Plone/public-folder/private-page from copy because user is not allowed to view the original."
